### DISCUSSION TITLE
Problem when compiling with PyInstaller

### DISCUSSION THREAD

**Discussion by Buzzardarg** [ibis-project/ibis#9040] [Q&A] [answered]
Hello there!

I'm having trouble compiling a script with PyInstaller and using Ibis. When running the program I get the error AttributeError: module 'ibis' has no attribute 'duckdb'.

I get that when using 'conn = ibis.duckdb.connect()' in my code.

I also tried using 'conn = ibis.connect("duckdb://") and in that case I get ValueError don't know how to connect to 'duckdb://'.

What's strange is that when I ran the script directly from Visual Studio Code either way, it runs without a problem. Any suggestions?

Thanks!

**Comment by lostmygithubaccount** [comment]
hi @Buzzardarg, I suspect you have [the other ibis](https://github.com/dmulholl/ibis) installed -- the PyPI package name for this Ibis is `ibis-framework`, which you would install with `pip install 'ibis-framework[duckdb]` (or your backend of choice)

can you confirm this is the issue? if you `pip list | grep ibis` and see `ibis` (even if you also see `ibis-framework`) that is likely the issue

**Reply by Buzzardarg** [reply]
Hi, thanks for answering!

I did as you said and there's no 'Ibis' package installed. I also tried pip uninstall ibis just in case and there was no such package.

[image](https://github.com/ibis-project/ibis/assets/87940062/af110ffd-ab0e-4077-90d3-67e4d71f2d7c)

I checked pyinstaller's log and I saw this lines:
missing module named 'duckdb.duckdb.typing' - imported by duckdb.typing (top-level)
missing module named 'duckdb.duckdb.functional' - imported by duckdb.functional (top-level)
missing module named ibis._ - imported by ibis (conditional), ibis.expr.visualize (conditional), ibis.expr.types.relations (delayed), ibis.expr.types.generic (delayed)
missing module named ibis.literal - imported by ibis (delayed), ibis.expr.types.relations (delayed)
missing module named 'ibis.expr.types.tvf' - imported by ibis.expr.types.relations (conditional)
missing module named 'sqlalchemy.base' - imported by duckdb_engine (conditional)
missing module named 'duckdb.duckdb.typing' - imported by duckdb.typing (top-level)
missing module named 'duckdb.duckdb.functional' - imported by duckdb.functional (top-level)
missing module named 'sqlalchemy.base' - imported by duckdb_engine (conditional)

Among others that make me suspect something is not quite right, but I thought that PyInstaller was suposed to bundle everything as long the packages are installed

**Reply by Buzzardarg** [reply, marked as answer]
I just tried copying some libraries from my venv (all folders starting with ibis, sqlalchemy and duckdb) to the dist folder and now it works, it seems that PyInstaller is indeed not bundling everything required. I do not know why though...
